Enable authentication for NATS Streaming (#3528)
kyma-project/kyma: resources/knative-provisioner-natss/README.md

Included releases:
 * https://github.com/knative/eventing/blob/v0.4.1/contrib/natss/config/provisioner.yaml

Kyma-specific changes:

> * The images are changed to custom ones created from `kyma-incubator/eventing`.

Reviewer: ```suggestion
* The images changed to custom ones created from `kyma-incubator/eventing`.
```

> * The images are changed to use the custom one we created from kyma-incubator/eventing.

Reviewer: ```suggestion
* The images are changed to custom ones created from `kyma-incubator/eventing`.
```

> * New environment variables: `EB_USER` and `EB_PASS` added for authentication.

Reviewer: ```suggestion
* New environment variables: **{EB_USER}** and **{EB_PASS}** added for authentication.
```

> * New environment variables `EB_USER` and `EB_PASS` are added for authentication.

Reviewer: ```suggestion
* New environment variables: `EB_USER` and `EB_PASS` added for authentication.
```

> * Environment variables `DEFAULT_CLUSTER_ID` and `DEFAULT_NATSS_URL` are set for Kyma-specific needs.

Reviewer: ```suggestion
* Environment variables **{DEFAULT_CLUSTER_ID}** and **{DEFAULT_NATSS_URL}** set for Kyma-specific needs.
```

> * A new label `rand` is added to the Deployments to force Pod restart during the upgrade.

Reviewer: ```suggestion
* A new label `rand` added to Deployments to force Pod restart during the upgrade.
```
